SConstruct

```diff
@@ -900,6 +909,33 @@ elif env['system_googletest'] == 'y':
 elif env['system_googletest'] == 'n':
     env['googletest'] = 'submodule'
 
+# Check for yaml-cpp library and checkout submodule if needed
+if env['system_yamlcpp'] in ('y', 'default'):
+    if conf.CheckCXXHeader('yaml-cpp/yaml.h', '""'):
+        env['system_yamlcpp'] = True
+        print("""INFO: Using system installation of yaml-cpp library.""")
+
+    elif env['system_yamlcpp'] == 'y':
+        config_error('Expected system installation of yaml-cpp library, but it '
+            'could not be found.')
+
+if env['system_yamlcpp'] in ('n', 'default'):
+    env['system_yamlcpp'] = False
+    print("""INFO: Using private installation of yaml-cpp library.""")
+    if not os.path.exists('ext/yaml-cpp/include/yaml-cpp/yaml.h'):
+        if not os.path.exists('.git'):
+            config_error('yaml-cpp is missing. Install source in ext/yaml-cpp.')
+
+        try:
+            code = subprocess.call(['git','submodule','update','--init',
```

Comment:
```suggestion
            code = subprocess.call(['git', 'submodule', 'update', '--init',
```

Reply:
Just wanted to try that feature out... nice. Next line too, if you feel like fixing it to be PEP 8

Reply:
Fixed - I didn't use the web UI thing so I could get both lines at once.